sdk/communication/communication-administration/test/utils/recordedClient.ts

```diff
@@ -90,19 +90,23 @@ export function createRecordedCommunicationIdentityClientWithToken(
     };
 
     return {
-      client: new CommunicationIdentityClient(env.COMMUNICATION_ENDPOINT, credential),
+      client: new CommunicationIdentityClient(endpoint, credential),
       recorder
     };
   }
 
   try {
-    credential = new DefaultAzureCredential();
+    credential = new ClientSecretCredential(
```

Comment:
Did DefaultAzureCredential not work as in Java and Python?

Reply:
This is an interesting one. Under the hood there isn't a huge difference between the two. But when debugging we found that using `DefaultAzureCredential` threw this complicated error that made it very hard to debug the issue, but when using `ClientSecretCredential` it gave a more specific error (telling us there was an issue with the env vars). That was why I wanted to stick to `ClientSecretCredential`. Thoughts @DominikMe ? I know you wanted to originally use  `DefaultAzureCredential` as well.

Reply:
@JoshuaLai Looking at your git history, you've switched to `ClientSecretCredential` in one commit which failed, then you added another commit that adds the missing env vars which passed. I don't think the switch to `ClientSecretCredential` was necessary, just the missing env vars were the culprit.

With the env vars present, `DefaultAzureCredential` should use `EnvironmentCreedential` which then uses `ClientSecretCredential`. So could you switch back to `DefaultAzureCredential` and try again? :-)